api/admin.go

```diff
@@ -195,6 +195,15 @@ func testEmail(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// if the user hasn't changed their email settings, fill in the actual SMTP password so that
+	// the user can verify an existing SMTP connection
+	if cfg.EmailSettings.SMTPServer == utils.Cfg.EmailSettings.SMTPServer &&
```

Comment:
Why are we checking that they didn't change anything else? What if they want to change the server or username but not the password?